libusb/descriptor.c

```diff
@@ -127,7 +126,7 @@ static int parse_endpoint(struct libusb_context *ctx,
 
 	/* Copy any unknown descriptors into a storage area for drivers */
 	/*  to later parse */
-	len = (int)(buffer - begin);
+	intptr_t len = (intptr_t)(buffer - begin);
```

Comment:
now, when I looke a this - what's the point of using `intptr_t` and not using `int` as it was originally?

Reply:
Admittedly, not much.

The old cast is theoretically truncating (64 bit pointers, 32 bit ints), but of course they should never be so far apart.

This way is just slightly more correct. `len` is now the correct type, and at least the check on the next line (`if (len <= 0)` is therefore more correct too in the face of overflow.

Reply:
When we casted to `int` right away, in case of overflow there was a chance it would be cut off by the `<0` check.
If we're to make it property right, then in addition to `<0` check we should also check for (at least) `INT_MAX`.

_Unless_ it is already checked elsewhere for some othe constant (e.g. something usb-specific), in which case casting it to `int` instead of `intptr_t` would make more elegant implementation.

Reply:
parse_configuration() or its sole caller raw_desc_to_config() is called with an _int_ size. raw_desc_to_config() is only called from libusb_get_active_config_descriptor() where the size is taken from wTotalLength which therefore cannot be larger than 65535 ever.

Reply:
What was the conclusion here, is the change pointless?

Reply:
My opinion is still that the change is a small improvement.

Reply:
I am fine with the change. But since `len` is now intptr_t there is no need for the cast, it is implicit on assignment.

Reply:
I've removed the cast, and also changed to `ptrdiff_t`, which is more correct.